src/components/auth0-login-zone/auth0-login-zone.js

```diff
@@ -4,25 +4,81 @@ import LoginSvgsAndText from "../login-zone/login-svgs-and-text"
 import routes from '../../routes'
 import withAuth0 from '../../auth/auth0-wrapper'
 import LoadingElement from '../generic-elements/loading-element'
+import { useState, useEffect, useRef } from 'react'
+import { Link } from 'react-router-dom'
+import FolderIcon from '../generic-elements/folder-icon'
+import ExitIcon from '@material-ui/icons/ExitToApp'
+import { isMobile } from "react-device-detect"
 
 const returnTo = `${window.location.origin}${routes.oauth}`
 
-export const Auth0LoginZone = ({ auth: { isAuthenticated, isLoading, loginWithRedirect, logout}  }) => {
+export const Auth0LoginZone = ({ auth: { isAuthenticated, isLoading, loginWithRedirect, logout } }) => {
+  const [isMenuToggled, setMenuToggled] = useState(false);
+  const [isMouseInMenu, setMouseInMenu] = useState(false);
+  const wrapperRef = useRef(null);
+  
+  const mouseEnterMenu = () => { if (!isMobile) {setMouseInMenu(true)}  }
+  const mouseExitMenu = () => { if (!isMobile) {setMouseInMenu(false)}  }
+
+  const toggleMenu = () => { if (isMobile) {setMenuToggled(!isMenuToggled)} }
+  const clickOutMenu = () => { if (isMobile) {setMenuToggled(false)} }
+  const isMenuExpanded = (isMenuToggled || isMouseInMenu)
+
+  useClickOutWatcher(wrapperRef, clickOutMenu);
+
   if (isLoading) {
     return <login-zone><LoadingElement /></login-zone>
   }
 
   return (
     <login-zone>
-      {
-        isAuthenticated
-          ? <button onClick={() => { logout({ returnTo }) }}><LoginSvgsAndText text="LOG OUT" /></button>
-          : <button data-testid="login-button" onClick={ loginWithRedirect }><LoginSvgsAndText text="LOG IN" /></button>
-      }
+      <div ref={wrapperRef}>
+        <div className="login-block">
+          {
+            isAuthenticated
+              ? <button className={isMenuExpanded ? "action" : "status"} onClick={toggleMenu} onMouseEnter={mouseEnterMenu} onMouseLeave={mouseExitMenu}>
+                <LoginSvgsAndText text="My Account" symbol={isMenuExpanded ? "▴" : "▾"} />
+              </button>
+              : <button className="action clickable" data-testid="login-button" onClick={loginWithRedirect}>
+                <LoginSvgsAndText text="Log in to your account" symbol={"▸"} />
+              </button>
+          }
+        </div>
+        {
+          (isAuthenticated && isMenuExpanded) &&
```

Comment:
Not sure I like how this is organized, but it is not quite unwieldy enough to be a problem